src/jobflow_remote/jobs/runner.py

```diff
@@ -1263,12 +1266,17 @@ def batch_get_process_id(
                 process_id=process_id,
             )
             return process_id
-        if self.job_controller.batches is not None:
-            doc = self.job_controller.batches.find_one(
-                {"batch_uid": batch_uid}, projection=["process_id"]
+        try:
+            process_id, wk_name = self.job_controller.get_batch_process_id(batch_uid)
+            if wk_name != worker_name:
+                raise RuntimeError("Wrong worker")
+        except MissingDocumentError:
+            process_id = None
+            logger.warning(
+                f"error trying to get the process id and worker for batch with unique id: {batch_uid}",
+                exc_info=True,
```

Comment:
I am not 100% sure I understand the implication of this. What would happen if batch_get_process_id returns None? Wouldn't it set the process_id to None and then never be set correctly again? Or will be updated again later?

Reply:
No that's True, it will not be updated again in the JobDoc. Actually, it will be updated first by update_running_jobs, and be set to None that time, and maybe - but I wouldn't know in which case this could really happen - it will be updated "correctly", i.e. with a real pid value, by the update_terminated_jobs. Once updated_terminated_jobs has been called, the job is set to TERMINATED and it will not try again. Maybe we could have something to update batch job ids manually in case something like this happens, e.g. an admin command or else that looks for jobs where remote.process_id is None, and tries to figure out from which batch it came from ?